### PR TITLE
Feature/entry version page

### PR DESCRIPTION
**Pull request by agduncan94**
* Reworking of the entry page (both my entries and public entries) to have versions as a top level selector
* Can link directly to specific versions of an entry
* On public pages will use a label to identify default version

### CODE REVIEW

**Comment by agduncan94** [comment]
I can give another quick demo if needed.

File: src/app/container/container.component.html [hunk 1]
```diff
@@ -116,19 +124,34 @@ <h3>
         </form>
       </tab>
       <tab heading="Versions" *ngIf="validVersions">
-        <app-versions-container [versions]="tool?.tags" id="versionsTab" [verifiedSource]="tool?.verifiedSources">
+        <app-versions-container [versions]="tool?.tags" id="versionsTab" [verifiedSource]="tool?.verifiedSources" [selectedVersion]="selectedVersion">
```
Comment (denis-yuen) [inline src/app/container/container.component.html]: This brings up an interesting question. How do we want verification to look if some versions are verified and some are not?

File: src/app/container/dockerfile/dockerfile.component.ts [hunk 2]
```diff
@@ -27,10 +27,17 @@ import { ContainerService } from '../../shared/container.service';
   selector: 'app-dockerfile',
   templateUrl: './dockerfile.component.html',
 })
-export class DockerfileComponent extends VersionSelector implements AfterViewChecked {
+export class DockerfileComponent implements AfterViewChecked {
 
   @Input() id: number;
   @Input() entrypath: string;
+  _selectedVersion: any;
```
Comment (denis-yuen) [inline src/app/container/dockerfile/dockerfile.component.ts]: Is it really not possible to narrow this down from `any` for these two lines?

File: src/app/container/launch/launch.component.ts [hunk 3]
```diff
@@ -17,41 +17,87 @@
...
+              private metadataService: MetadataService) {
+    this.metadataService.getDescriptorLanguages().subscribe(map => {
+      this.descriptors = map;
+      this.validDescriptors = this.filterDescriptors(this.descriptors, this._selectedVersion);
+    });
   }
-  getDescriptors(currentVersion): any {
-    return this.toolDescriptorService.getDescriptors(this.versions, this.default);
+
+  filterDescriptors(descriptors: Array<any>, version: any): Array<any> {
```
Comment (denis-yuen) [inline src/app/container/launch/launch.component.ts]: a bunch of `any` here too

**Comment by agduncan94** [comment]
@garyluu Could you check this over

File: src/app/container/container.component.html (hunk 1 above)
Comment (agduncan94) [reply src/app/container/container.component.html]: Currently we have the following:
* Checkmark beside entry name to show that at least one version is verified
* In versions tab, column to show which versions are verified
* In side bar for public viewing, we have a box with the verified versions

File: src/app/container/container.component.ts [hunk 4]
```diff
@@ -68,7 +70,7 @@ export class ContainerComponent extends Entry {
     router: Router,
     private containerService: ContainerService,
     stateService: StateService,
-    errorService: ErrorService) {
+    errorService: ErrorService, route: ActivatedRoute) {
```
[the remark this answers is not in the document]
Comment (agduncan94) [reply src/app/container/container.component.ts]: oops that should be deleted

File: src/app/container/container.component.ts [hunk 5]
```diff
@@ -240,4 +255,8 @@ export class ContainerComponent extends Entry {
     this.containerService.setCopyBtn(copyBtn);
   }
 
+  onTagChange(tag): void {
```
Comment (garyluu) [inline src/app/container/container.component.ts]: See if tag could have a certain event type.

File: src/app/workflow/workflow.component.ts [hunk 6]
```diff
@@ -53,10 +53,12 @@ export class WorkflowComponent extends Entry {
   private workflowSubscription: Subscription;
   private workflowCopyBtnSubscription: Subscription;
   private workflowCopyBtn: string;
+  public selectedVersion = null;
+  public urlVersion = null;
   constructor(private dockstoreService: DockstoreService, dateService: DateService, private refreshService: RefreshService,
     private workflowsService: WorkflowsService, trackLoginService: TrackLoginService, providerService: ProviderService,
     router: Router, private workflowService: WorkflowService,
-    stateService: StateService, errorService: ErrorService) {
+    stateService: StateService, errorService: ErrorService, route: ActivatedRoute) {
```
Comment (garyluu) [inline src/app/workflow/workflow.component.ts]: Same for here

**Pull request merged**